nss-day-cohort-60/national-parks-api
# JD added create/update/delete for users, user_favorites, amenities, park_amenities

# Description 
Created functions for PUT, POST, DELETE for users, amenities, park_amenities. User_Favorites do not need PUT.

# Steps to test user functions:

1. Fetch the JD-CRUD-Users-Amenities branch and switch to it
2. Start/restart debugger
3. Send a POST request to Users (e.g. http://localhost:8088/users) 
post body:     {
        "first_name": "Test",
        "last_name": "Post",
        "email": "tests@gandalfsucks.com",
        "password": "gandalfsucks12",
        "isRanger": 0
    }
4. Verify 201 status code and new User in list of Users
5. Send a PUT request to http://localhost:8088/users/{new_post_id} 
with post body:  {
        "first_name": "Toast",
        "last_name": "Bread",
        "email": "breakfast@gandalfsucks.com",
        "password": "gandalfsucks12",
        "isRanger": 0
    }
5. Verify changes to that user id with a GET request. PUT requests default to 204 status, needs a new ticket
6. Send a DELETE request to http://localhost:8088/users/{new_post_id} 
7. Verify 204 status code and the user is no longer in the Users list
8. Repeat process for:
     - http://localhost:8088/user_favorites (should not PUT)
       post body:  {
        "type_id": 2,
        "post_id": 7,
        "user_id": 11
    }
     - http://localhost:8088/amenities
       post body:  { "type": "Karaoke" }
     - http://localhost:8088/park_amenities (should test name: null and name: "string" for thoroughness)
        post body:      {
        "name": null,
        "amenity_id": 2,
        "park_id": 1
    }
